<launch>
  <!-- Arguments -->
  <arg name="map_file" default="$(find testbot)/maps/map.yaml"/>
  <arg name="open_rviz" default="true"/>
  <arg name="move_forward_only" default="false"/>
  <arg name="init_x_" default="0.932" />
  <arg name="init_y_" default="2.75" /> <!--2.68-->
  <arg name="init_yaw_" default="-1.5707963" />

  <node pkg="tf" type="static_transform_publisher" name="map_odom_broadcaster"  args="0 0 0 0 0 0 map odom 100"/>

  <!--for obstacle layer (lidar)-->
  <node name="base_to_laser" pkg="tf2_ros" type="static_transform_publisher" args="0 0 0 0 0 -0.044 0.999 base_footprint laser_frame"/>

  <!--Publish odometry-->
  <node pkg="testbot" name = "odom_integral" type = "odom_integral"  output="screen">
    <param name="init_x" value="$(arg init_x_)"/>
    <param name="init_y" value="$(arg init_y_)"/>
    <param name="init_yaw" value="$(arg init_yaw_)"/>
  </node>
  <!-- <node pkg="testbot" name = "odometry_test_old" type = "odometry_test_old"  output="screen">
    <param name="init_x" value="$(arg init_x_)"/>
    <param name="init_y" value="$(arg init_y_)"/>
    <param name="init_yaw" value="$(arg init_yaw_)"/>
  </node>  -->
  <!-- <node pkg="testbot" name = "odometry_test" type = "odometry_test"  output="screen">
    <param name="init_x" value="$(arg init_x_)"/>
    <param name="init_y" value="$(arg init_y_)"/>
    <param name="init_yaw" value="$(arg init_yaw_)"/>
  </node> -->

  <!--Change cmd_vel to txST1-->
  <node pkg="testbot" name = "planner_to_ST" type = "planner_to_ST"  output="screen">
    <param name="init_x" value="$(arg init_x_)"/>
    <param name="init_y" value="$(arg init_y_)"/>
    <param name="init_yaw" value="$(arg init_yaw_)"/>
  </node>
  <!-- <node pkg="testbot" name = "planner_to_ST_modify" type = "planner_to_ST_modify"  output="screen">
    <param name="init_x" value="$(arg init_x_)"/>
    <param name="init_y" value="$(arg init_y_)"/>
    <param name="init_yaw" value="$(arg init_yaw_)"/>
  </node> -->

  <!--For cup input message-->
  <!-- <node pkg="cup_layer" name = "cup_server" type = "cup_server"  output="screen" />-->
  <!--<node pkg="cup_layer" name = "cup_publisher" type = "cup_publisher"  output="screen" />-->

  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

  <!-- move_base -->
  <include file="$(find testbot)/launch/move_base.launch">
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
  </include>

  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true" args="-d $(find testbot)/rviz/turtlebot3_navigation.rviz"/>
    <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find teb_local_planner_tutorials)/cfg/rviz_navigation.rviz"/> -->
  </group>

</launch>
